### ISSUE TITLE
Datadog-cluster-agent is not given secrets read rights

### ISSUE THREAD

**Issue by frankbranham** [DataDog/helm-charts#479]
Trying out the new 7.32.2 k8s secrets ENC support by setting:
DD_SECRET_BACKEND_COMMAND=/readsecret_multiple_providers.sh and the cluster shows the error. 

User "system:serviceaccount:ops:datadog-cluster-agent" cannot get resource "secrets" in API group "

Expected:
ENC[] secrets to work. 

Looking at the cluster role the chart creates for datadog-cluster-agent in my cluster, there is no block for secrets defined. 

Using k8s Helm chart  version: 2.27.7

**Comment by ahmed-mez** [comment]
Hi @frankbranham 

Thanks for opening this issue and sorry for the trouble.

We acknowledge the issue and we'll try to get this fixes as soon as we can. Ideally, we won't be adding rbacs to get secrets unless `DD_SECRET_BACKEND_COMMAND=/readsecret_multiple_providers.sh` is configured.

**Comment by ahmed-mez** [comment]
hey @frankbranham 

The agent will have the read rights now if you use `datadog.secretBackend.command="/readsecret_multiple_providers.sh"` to configure your secret backend. Please make sure you pull the latest helm chart version as well.

**Comment by frankbranham** [comment]
Still not working with 2.22.13 and DD_SECRET_BACKEND_COMMAND=/readsecret_multiple_providers.sh and the cluster shows the error.

I don't see any change in my datadog_cluster_agent role.

**Comment by jackhodkinson** [comment]
I have tried the above and still having issues.
- I have the latest helm chart (2.28.14) 
- I have set `datadog.secretBackend.command` to `"/readsecret_multiple_providers.sh"`.
- I am using a secret in the `clusterAgent.confd` section of the values file.
- The logs from the cluster agent show the following error:

```
2022-01-25 17:24:26 UTC | CLUSTER | ERROR | (pkg/autodiscovery/autoconfig.go:305 in processNewConfig) | Dropping conf for 'postgres': error while decrypting secrets in an instance: invalid executable '/readsecret_multiple_providers.sh': can't stat it: no such file or directory
```

Looks like the agent is reading the `datadog.secretBackend.command` variable correctly but this file is not found in the cluster agent's image?

**Comment by ahmed-mez** [comment]
@jackhodkinson sorry for the trouble! what is your cluster agent version? This feature requires cluster agent 1.16.0

**Comment by jackhodkinson** [comment]
That fixed it 🎉

As you suspected the cluster agent version was not up to date. I was on version 1.12.0. I did not realize you had to upgrade the cluster agent separately to the chart 🤦

Fixed by bumping the version to 1.16.0 (by setting `clusterAgent.image.tag` in the values file.

Thanks!
